File: data estructures/stack.py
""" 
This algorithm implements a generic version of a Stack in python, the idea
with the algorithm in this directory is have a source for future reference in
problems from codility or similar sites/problems.

The algorithm used the theory presented in Introduction to Algorithms, 
a book by Thomas H. Cormen.

@author: Iara Ribeiro - 11/12/2016.
""" 
class Stack():
	
	def __init__(self):
		self.top = 0
		self.S = []

	def stackEmpty(self):
		if(self.top == 0):
			return "true"
		return "false"
	
	def push(self,x):
		self.top= self.top + 1
		self.S.insert(self.top, x)

	def pop(self):
		if(self.top == 0):
			return "underflow"
		else:
			self.S.remove(self.S[(self.top-1)])
			self.top = self.top - 1
	
	def print_stack(self):
		result = "[";
		for i in range(0,self.top):
			if(i==0):
				result += self.S[i]
			else:
				result = result + "," + self.S[i]
		return result+"]"

	def test(functionToTest, args, expected):
		result = functionToTest(args)
		print "Expected " + expected + ", was " + result		

if __name__ == '__main__':
    s = Stack()
    print("Expected " + s.stackEmpty() + ", was " + "true")
    s.push("banana")
    print("Expected " +"[banana]" + ", was " + s.print_stack())
    s.push("apple")
    print("Expected " +"[banana,apple]"+ ", was " +s.print_stack())
    s.pop()
    print("Expected " +"[banana]" + ", was " + s.print_stack())    
    s.push("pineapple")
    print("Expected " +"[banana,pineapple]"+ ", was " +s.print_stack())
    s.pop()
    s.pop()
    print("Expected " +"[]"+ ", was " +s.print_stack())

File: data estructures/stack.js
/*
This algorithm implements a generic version of a Stack in javascript, the idea
with the algorithm in this directory is have a source for future reference in
problems from codility or similar sites/problems.

The algorithm used the theory presented in Introduction to Algorithms, 
a book by Thomas H. Cormen.

@author: Iara Ribeiro - 11/12/2016.
*/
var top = 0;
var S = [];

function stackEmpty(S){
	if(top === 0){
		return true;
	}
	return false;
}

function push(S,x){
	top += 1;
	S[top] = x;
}

function pop(S){
	if(stackEmpty(S)){
		return "stack empty";
	}else{
		top -= 1;
		return S[top + 1];
	}
}

function print(S){
	var result = "[";
	for(var i = 1; i <= top; i++){
		if( i === 1){
			result += S[i];
		}else{
			result = result + "," + S[i];
		}
	}
	return result+"]";
}

function test(functionToTest, args, expected) {
	var result = functionToTest.apply(this, args);
    console.log("Expected " + expected + ", was " + result);
}

test(stackEmpty, S, true);
push(S,"banana");
test(print, [S], "[banana]");
push(S,"apple");
test(print, [S], "[banana,apple]");
pop(S);
test(print, [S], "[banana]");
push(S,"pineapple");
test(print, [S], "[banana,pineapple]");
pop(S);
test(print, [S], "[]");